src/overmapbuffer.h

```diff
@@ -54,6 +54,15 @@ struct city_reference {
     int get_distance_from_bounds() const;
 };
 
+/**
+* Container for a pointer to an overmap and a set of coordinates which are
+* local to that overmap.
+*/
+struct overmap_reference_with_local_coordinates {
+    overmap *overmap;
```

Comment:
GCC is deeply unhappy about shadowing a class name with a variable name.
```suggestion
    overmap *overmap_pointer;
```